src/LanguageServer/Protocol/Handler/Completion/CompletionResultFactory.cs

```diff
@@ -18,23 +18,27 @@
 using Microsoft.CodeAnalysis.Shared.Collections;
 using Microsoft.CodeAnalysis.Shared.Extensions;
 using Microsoft.CodeAnalysis.Text;
+using Roslyn.Text.Adornments;
 using Roslyn.Utilities;
 using LSP = Roslyn.LanguageServer.Protocol;
 
 namespace Microsoft.CodeAnalysis.LanguageServer.Handler.Completion
 {
-    internal abstract class AbstractLspCompletionResultCreationService : ILspCompletionResultCreationService
+    internal static class CompletionResultFactory
```

Comment:
📝 This is essentially a merge of [AbstractLspCompletionResultCreationService](https://github.com/dotnet/roslyn/blob/main/src/LanguageServer/Protocol/Handler/Completion/AbstractLspCompletionResultCreationService.cs), [DefaultLspCompletionResultCreationService](https://github.com/dotnet/roslyn/blob/main/src/LanguageServer/Protocol/Handler/Completion/DefaultLspCompletionResultCreationService.cs), and [EditorLspCompletionResultCreationService](https://github.com/dotnet/roslyn/blob/main/src/EditorFeatures/Core/LanguageServer/EditorLspCompletionResultCreationService.cs)